File: pset6/readability/readability.py
from cs50 import get_string

sentences = 0
words = 0
letters = 0
s = get_string("Text: ")
for i in s:
    if i == "." or i == "?" or i == "!":
        sentences+=1
    elif i == " ":
        words+=1
    elif (i>="a" and "z">=i) or ("Z">=i and i>="A"):
        letters+=1
words+=1
index=0.0588 * 100*(letters/words) - 0.296 * 100*(sentences/words) - 15.8
index=int(round(index, 0))
if index >= 16:
    print("Grade 16+")
elif index <1:
    print("Before Grade 1")
else:
    print("Grade", index)

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
int main(void)
{
    string s = get_string("Text: ");
    int len = strlen(s);

    int letters = 0;
    int wordcount=  0;
    int sentences= 0;

    for(int i = 0; i<len;i++)
    {
        if(s[i]==46||s[i]==33||s[i]==63){
            sentences++;
        }
        if(s[i]==32){
            wordcount++;
        }
        else if(s[i]>=97&&s[i]<=122){
            letters++;
        }
        else if(s[i]>=65&&s[i]<90){
            letters++;
        }
        else{}
    }
    wordcount++;
    double index = 0.0588 * letters/wordcount*100 - 0.296 * sentences/wordcount*100 - 15.8;
    int grade = 0;
    if(index-(int)index>.5){
        grade = (int)index+1;
    }
    else{
        grade = (int) index;
    }
    if(grade<1){
        printf("Before Grade 1\n");
    }
    else if(grade>15){
        printf("Grade 16+\n");
    }
    else{
    printf("Grade %i\n", grade);
    }
}